app/scripts/models/verification/same-browser.js

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Verification info that is stored locally in case the user verifies
+ * in the same browser. This allows information like `context`
+ * to be re-used if the user verifies in the same browser, but not
+ * in a different browser.
+ *
+ * Information is persisted into localStorage. Either an `email` or `uid`
+ * is required for the verification info to be fetched.
+ *
+ * `uid` is used in signup and account unlock verifications
+ * `email` is used in password reset verifications.
+ */
+
+define([
+  'backbone',
+  'lib/storage'
+], function (Backbone, Storage) {
+  'use strict';
+
+  var STORAGE_KEY = 'verificationInfo';
+
+  var proto = Backbone.Model.prototype;
+
+  var SameBrowserVerificationInfo = Backbone.Model.extend({
+    initialize: function (attributes, options) {
```

Comment:
I just found out about the `attribute, options` paradigm used by backbone. We conflate the two everywhere, which leads to a horrible situation where non-data options end up in attributes. We should fix this in a follow up issue.